chacractl/api/binaries.py

```diff
@@ -71,27 +86,44 @@ def post(self, url, filepath):
             return self.put(file_url, filepath)
         elif exists.status_code == 404:
             logger.info('POSTing file: %s', filepath)
-            with open(filepath, 'rb') as binary:
-                response = requests.post(
+            binary, digest = self.load_file(filepath)
+            response = requests.post(
                     url,
                     files={'file': binary},
                     auth=chacractl.config['credentials'],
                     verify=chacractl.config['ssl_verify'])
         if response.status_code > 201:
             logger.warning("%s -> %s", response.status_code, response.text)
             response.raise_for_status()
+        if not self.verify_upload(url, filename, digest):
+            # Since this is a new file, attempt to delete it
+            logging.error(
+                    'Checksum mismatch: server has wrong checksum for %s! '
+                    'Deleting corrupted file from server...',
+                    filepath)
+            self.delete(file_url)
+            raise SystemExit(
```

Comment:
If we retry, how many times should we retry? It should probably be configurable.

If we continue onto other files, the exit status should still be non-zero, to indicate that a failure occurred. At least the operation will be idempotent. I thought it was best to avoid confusion by failing on the first mismatch, since the user could try again by just rerunning the command, and the failure should halt any automation around this operation.